### ISSUE TITLE
For GD77 users

### ISSUE THREAD

**Issue by denhamgithub** [sergev/dmrconfig#25]
The Radioddity GD77 radio has had quite a few firmware update/upgrades.  And some of these, unfortunately, have required the user to re-build their GD77 codeplug from the ground up.  For example, see the documentation for the latest firmware 3.1.08.

dmrconfig simplified the whole process, saving me a whole evening.  The key is use dmrconfig to write a conf text file before you decide to upgrade to new firmware.  If the new firmware revision from Radioddity requires you to re-build your codeplug from the ground up, you can upgrade the firmware.  And use dmrconfig -c option to write the text file from the older firmware version back to the new firmware.
Dennis Eckert KB7ST

**Comment by sergev** [comment]
That's right: migration is much easier with dmrconfig. It's even possible to convert codeplugs across different types of radios. I know people were able to transform MD-380 images into D868UV configuration, and adapt GD-77 data for RD-5R.

**Comment by tylert** [comment]
@sergev Is this procedure for working with "converted/transformed codeplugs" in dmrconfig documented somewhere?  How easy it is to work with codeplugs that are supported by Editcp (https://github.com/DaleFarnsworth-DMR/editcp/issues/5) and then use dmrconfig to convert+upload them to a radio?  Are there any likely "feature bugs" with this method other than ones like https://github.com/sergev/dmrconfig/issues/43 (if using the OpenGD77 firmware;  https://github.com/rogerclarkmelbourne/OpenGD77)?

At the moment, I only own two TYT MD-390G UHF-only radios running md380tools firmware (https://github.com/travisgoodspeed/md380tools;  which are somewhat uninteresting radios to use in a rural area with lots of VHF FM repeaters, spotty internet and no local DMR repeaters).  I don't yet own a Radioddity GD-77 (a.k.a. TYT MD-760?) to try things out and have been reluctant to jump in and buy a few to run OpenGD77 firmware and use them as "daily drivers" without knowing for sure that I can use a command-line-friendly JSON<->codeplug workflow that "runs well" on Linux (in containers;  such as my ugly https://github.com/tylert/freq-lists/tree/master/codeplugs junk).  @rogerclarkemelbourne do you have any sage advice?

I'm very new to the world of codeplugs and DMR and am still in the process of learning Go more and helping with issues such as this one would be an ideal opportunity for me if someone can please confirm for me that the possibility of using the combination of dmrconfig, Editcp and OpenGD77 (as an alternative to their community CPS) is actually possible on Linux.  I would love to purchase a few Radioddity GD-77s and try to help out.

Tyler VA3DGN
